modules/standard/IO.chpl

```diff
@@ -1518,7 +1518,7 @@ proc open(path:string="", mode:iomode, hints:iohints=IOHINT_NONE,
     var host_start = path.find("//") + 2;
     var colon = path.find(":", host_start..);
     var slash = path.find("/", host_start..);
-    var host_end, port_start, port_end, path_start = 0;
+    var host_end, port_start, port_end, path_start: byteIndex = 0;
```

Comment:
Are there other places in module code using the codepoint indexing now where we might prefer it to use byte indexing? Did you look? Can you think of a way to check?

Reply:
I've updated Regexp.chpl as you suggested.  I mostly looked for other places by looking for uses of `string.find()` and related things, which is why I missed Regexp.

I think I've updated enough module code to use `byteIndex` for now.  We can look again if we find performance effects.  The issue is that updating a module to use one of the index types turns out to be a lot of work with multiple iterations, because the index types are viral like `const`, and once they are used, eventually there comes a point where a decision has to be made about where to cast back to `int` and be rid of them.